components/docker-auth/cmd/authn/authentication.go

```diff
@@ -43,237 +43,268 @@ const (
 	logFile               = "/extension-logs/authentication.log"
 )
 
-func readCert(certPathEnv string) ([]byte, error) {
+func readCert(certPathEnv string, execId string) ([]byte, error) {
 	key, err := ioutil.ReadFile(os.Getenv(certPathEnv))
 	if err != nil {
+		log.Printf("[%s] Unable to read the cert : %s\n", execId, err)
 		return nil, err
 	}
+	log.Printf("[%s] Read cert successfully\n", execId)
 	return key, nil
 }
 
-func getJWTClaims(token string) jwt.MapClaims {
+func getJWTClaims(token string, execId string) jwt.MapClaims {
 	jwtToken, _ := jwt.Parse(token, nil)
 	claims, ok := jwtToken.Claims.(jwt.MapClaims)
 	if ok {
+		log.Printf("[%s] Received JWT claims successfully\n", execId)
 		return claims
 	}
 	return nil
 }
 
-func getClaimValue(claim jwt.MapClaims, claimKey string) string {
+func getClaimValue(claim jwt.MapClaims, claimKey string, execId string) string {
 	value, ok := claim[claimKey].(string)
 	if ok {
+		log.Printf("[%s] Received JWT claim for the claim key %s successfully\n", execId, claimKey)
 		return value
 	}
 	return ""
 }
 
-func validateToken(inToken string, cert []byte) (bool, error) {
+func validateToken(inToken string, cert []byte, execId string) (bool, error) {
 	publicRSA, err := jwt.ParseRSAPublicKeyFromPEM(cert)
 	if err != nil {
+		log.Printf("[%s] Error parsing the cert : %s\n", execId, err)
 		return false, err
 	}
 	token, err := jwt.Parse(inToken, func(token *jwt.Token) (interface{}, error) {
 		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
-			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+			return nil, fmt.Errorf("[%s] Unexpected signing method: %s\n", execId, token.Header["alg"])
 		}
 		return publicRSA, err
 	})
 	if token != nil && token.Valid {
+		log.Printf("[%s] Token received is valid\n", execId)
 		return true, nil
 	}
+	log.Printf("[%s] Token received is invalid\n", execId)
 	return false, err
 }
 
 func main() {
+	err := os.MkdirAll("/extension-logs", os.ModePerm)
+	if err != nil {
+		log.Println("Error creating the file :", err)
+	}
 	file, err := os.OpenFile(logFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
 	if err != nil {
-		log.Fatalf("Error opening file: %v", err)
+		log.Printf("Error opening file: %s\n", err)
```

Comment:
You can move this log inside the if condition in line 111 and remove this if block